utils/updateFileApproval.ts

```diff
@@ -0,0 +1,35 @@
+import { Student } from "@prisma/client";
+import { mutate } from "swr";
+
+/**
+ * updateFileApproval updates a file's approval from an admin
+ * @param filePath
+ * @param id
+ * @param approval
+ * @returns deleted records
+ */
+export async function updateFileApproval(
+    filePath: string,
+    id: number,
+    approval: boolean | null,
+): Promise<Response> {
+    const updateData = {
+        id: id as number,
+        approval: approval,
+    };
+
+    const request = {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify(updateData),
+    };
+
+    const response = await fetch(
+        `/api/admin/update-${filePath}-status`,
```

Comment:
@ZhangCreations  hmm this is an interesting way of doing things. Do you think it'd make sense to add one more level of abstraction here and make a updateCriminalCheckApproval, updateIncomeProofApproval, etc... which uses this util? That way you don't have to keep a random FILE_PATH const in the main page.  Keeping stuff contained 👍 

So, instead of using updateFileApproval in your page, you'll just use updateCriminalCheckApproval instead, which calls this util indirectly.

Reply:
I think I personally like the modularity of this component more, it makes less files and maintaining the FILE_PATH const in the top level isn't too much additional data. Up to you however, let me know if I should change this.

Reply:
I like this util as well, it's clean. It's just that stating the FILE_PATH through multiple pages could still be redundant and I'd rather call `updateCriminalCheckApproval(id, true)` vs something like `updateFileApproval("criminal-check", id, true)` everytime. I was thinking we can make another function that we can export in this file. Something like


```typescript
export async function updateCriminalCheckApproval(
    id: number,
    approval: boolean | null,
): Promise<Response> {
    return updateFileApproval("criminal-check", id, approval)
}
```

it's not much additional code and it goes a long way 😄 ,

Reply:
If I make a function `updateCriminalCheckApproval` for the fileDownload card it makes it so that the fileDownloadCard component is no longer reusable for the proof of income page.

Reply:
That's true. My judgement is that we probably don't want to be passing slightly arbitrary strings around. Maybe we can even create an enum of some kind then I'd be happy with passing the enum around instead. Maybe for now, do you think It'd make sense to specify the type of file path as `"criminal-check" | "income-proof"`?